### PR TITLE
move backend stores into their own package

### PR DESCRIPTION
**Pull request by woodsaj**
As backend stores are pluggable, they should be in their own package.
Otherwise, their global variables get initilized when mdata is loaded.

### CODE REVIEW

File: cmd/mt-store-cat/full.go [hunk 1]
```diff
@@ -7,28 +7,28 @@ import (
...
 )
 
-func chunkSummary(ctx context.Context, store *mdata.CassandraStore, tables []string, metrics []Metric, keyspace, groupTTL string) error {
+func chunkSummary(ctx context.Context, store *cassandra.CassandraStore, tables []string, metrics []Metric, keyspace, groupTTL string) error {
 	now := uint32(time.Now().Unix())
-	end_month := now - (now % mdata.Month_sec)
+	end_month := now - (now % cassandra.Month_sec)
 
 	for _, tbl := range tables {
 		// actual TTL may be up to 2x what's in tablename. see mdata/store_cassandra.go for details
```
Comment (replay) [inline cmd/mt-store-cat/full.go]: i think that filename in the comment needs updating now

File: store/cassandra/store_cassandra.go [hunk 2]
```diff
@@ -80,12 +81,15 @@ var (
 	errmetrics = cassandra.NewErrMetrics("store.cassandra")
 )
 
-/*
-https://godoc.org/github.com/gocql/gocql#Session
-Session is the interface used by users to interact with the database.
-It's safe for concurrent use by multiple goroutines and a typical usage scenario is to have one global session
-object to interact with the whole Cassandra cluster.
-*/
+type ChunkReadRequest struct {
```
Comment (replay) [inline store/cassandra/store_cassandra.go]: why is the definition of `ChunkReadRequest` moved here, but `ChunkWriteRequest` remains where it was?

Comment (woodsaj) [reply store/cassandra/store_cassandra.go]: `chunkWriteRequest`s are used in the `store` interfaces `Add` method so all backend stores use them. The `chunkReadRequest` is a Cassandra specific implimentation detail

**Comment by Dieterbe** [comment]
what's the reason for adding the drop functionality to the mock store while also keeping devnullstore?
shouldn't we either add drop support to mock and deprecate devnull OR keep using devnullstore (and in that case, no need to add drop support to mock)

**Pull request merged**